### PR TITLE
list-permissions command

### PR DESCRIPTION
**Pull request by bookdude13**
Fulfills #9. Extends current list-permissions command with an optional user private key to get a specific user's permissions, not just the list of all permissions.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -118,10 +118,17 @@ fn run_remove_sequence(args: Args) -> Result<(), Error> {
...
 fn run_list_permissions(args: Args) -> Result<(), Error> {
-	let permissions = proton_cli::get_permissions();
-	Ok(println!("{}", permissions.join("\n")))
+	let private_key = args.arg_private_key;
+	if private_key.is_some() {
+		let _ = proton_cli::get_permissions(&private_key.unwrap())
+			.map(|p| println!("{:?}", p));
+		Ok(())
+	} else {
+		let permissions = proton_cli::get_permissions_list()
```
Comment (rjayatilleka) [inline src/main.rs]: Is this even necessary? I don't really understand why we need a list of the actual permissions. Where would we use that?

File: src/main.rs [hunk 2]
```diff
@@ -118,10 +118,17 @@ fn run_remove_sequence(args: Args) -> Result<(), Error> {
...
 }
 
-#[allow(unused_variables)]
 fn run_list_permissions(args: Args) -> Result<(), Error> {
-	let permissions = proton_cli::get_permissions();
-	Ok(println!("{}", permissions.join("\n")))
+	let private_key = args.arg_private_key;
+	if private_key.is_some() {
+		let _ = proton_cli::get_permissions(&private_key.unwrap())
+			.map(|p| println!("{:?}", p));
```
Comment (rjayatilleka) [inline src/main.rs]: What does this debug print actually look like? Can you show an example output, exactly as it would be printed?

**Comment by rjayatilleka** [comment]
Ready for response.

File: src/main.rs (hunk 1 above)
Comment (bookdude13) [inline src/main.rs]: The idea was that people unfamiliar with the project (or us, once we forget what we did) could easily list the permissions they could use in the set-permission command. I don't think we list them anywhere else

Comment (rjayatilleka) [inline src/main.rs]: Yeah, but we can just look at the code. Or look in the GUI when someone else is doing it. It's not like people will need to write in the permission and target manually, they'll just be choosing from a couple dropdowns.

Comment (bookdude13) [inline src/main.rs]: True. Also, I just discovered that the valid permission types are indeed printed if an invalid one is given. Will remove.

File: src/main.rs (hunk 2 above)
Comment (bookdude13) [inline src/main.rs]: It's ugly right now. Thoughts on how you would like to view it? Or should we just return the Vec<Permissions> and let the caller decide how to format it?
`[Permission { which: Administrate, target: None }, Permission { which: EditSeq, target: Some("asdf") }, Permission { which: EditSeqSec, target: Some("glory,1") }]`

Comment (rjayatilleka) [inline src/main.rs]: Can you JSON-ify it?

Comment (bookdude13) [inline src/main.rs]: Good idea. Done.

**Comment by bookdude13** [comment]
Done with revisions

**Comment by rjayatilleka** [comment]
LGTM

**Pull request merged**